pkgs/by-name/do/docker-init/package.nix

```diff
@@ -0,0 +1,46 @@
+{
+  stdenv,
+  fetchurl,
+  zstd,
+  lib,
+}:
+stdenv.mkDerivation rec {
+  pname = "docker-init";
+  version = "v1.3.0";
+  tag = "157355";
+
+  src = fetchurl {
+    url = "https://desktop.docker.com/linux/main/amd64/${tag}/docker-desktop-x86_64.pkg.tar.zst";
+    hash = "sha256-ysZorPBmoUvTJFFKDbZgQxPamONJKcXezmMrpZSVpwY=";
+  };
+
+  nativeBuildInputs = [
+    zstd
+  ];
+
+  unpackPhase = ''
+    runHook preUnpack
+    tar --zstd -xvf $src
+    runHook postUnpack
+  '';
+
+  installPhase = ''
+    runHook preInstall
+    mkdir -p $out/{bin,libexec/docker/cli-plugins}
+
+    cp usr/lib/docker/cli-plugins/docker-init $out/libexec/docker/cli-plugins
+    ln -s $out/libexec/docker/cli-plugins/docker-init $out/bin/docker-init
+    runHook postInstall
+  '';
+
+  meta = with lib; {
```

Comment:
`with lib` has been deprecated #208242

Reply:
Then, would
```nix
meta = let
  inherit (lib) licenses sourceTypes maintainers;
in {
  # ...
};
```
be a better fit, or should I reference them directly (e.g. `with lib.sourceTypes;`)?

Reply:
Either removes the potential for ambiguity (e.g. with fields like `version`), but I personally believe the latter, ie

```
meta = {
    license = lib.licenses.unfree;
    ....
```

is clearest